### PR TITLE
Deep fryer

### PR DESCRIPTION
**Pull request by Time-Green**
Ports https://github.com/tgstation/tgstation/pull/23618 by @iamgoofball
the mans good
:cl: iamgoofball(portedbytimegreen)
rscadd: Adds a deep fryer to kitchen
/:cl:

I'll add the wrench paths later once I get around to porting coiax's construction rework

### CODE REVIEW

**Comment by Kmc2000** [comment]
Yes

**Comment by matskuman5** [comment]
Rob Logan bites the deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried monkey meat cutlet.

**Comment by Time-Green** [comment]
I'm sorry matsk, I tried, but it's impossible

**Comment by Kmc2000** [comment]
:-1: do not merge until you can make the "lightly fried deep fried lightly fried lightly fried lightly fried chicken tendies"

**Comment by Cruix** [comment]
Time's skills are not up to the legendary Doublefry technique

**Comment by Time-Green** [comment]
Deep fried deep fried deep fried nuke disc it is then

**Comment by drderp3635** [comment]
Maybe make it so upgrading the laser will increase nutriment added to the food, and decrease amount used

**Comment by Time-Green** [comment]
Actually I checked the construction rework and that would quite literally kill me

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/*
+April 3rd, 2014 marks the day this machine changed the face of the kitchen on NTStation13
```
Comment (Super3222) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:2]: the fuck

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+			frying = null
+			cook_time = 0
+			return
+	else if(user.pulling && user.a_intent == "grab" && iscarbon(user.pulling) && reagents.total_volume)
+		if(user.grab_state < GRAB_AGGRESSIVE)
+			user << "<span class='warning'>You need a better grip to do that!</span>"
+			return
+		var/mob/living/carbon/C = user.pulling
+		user.visible_message("<span class = 'danger'>[user] dunks [C]'s face in [src]!</span>")
+		reagents.reaction(C, TOUCH)
```
Comment (Super3222) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:117]: What'd be cool is if you added some sort of hissing sound after this

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm (hunk 1 above)
Comment (Time-Green) [reply code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:2]: No idea, I never question goofball

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm (hunk 2 above)
Comment (Time-Green) [reply code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:117]: Good idea, I'll go grab some sounds when I can

**Comment by ShadowDeath6** [comment]
you spelled goofball wrong in your changelog author thing

**Pull request merged**